src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -274,9 +387,20 @@ extern "C" int CallApplicationMain(CALL_APPLICATION_MAIN_DATA* data)
         return 1;
     }
 
-    setenv("DNX_FRAMEWORK", "dnxcore50", 1);
+    int result = 0;
 
-    auto result = CallMain(data);
+    MultiByteToWideChar = (MultiByteToWideChar_fn)dlsym(pLibCoreClr, "MultiByteToWideChar");
+    if (!MultiByteToWideChar)
+    {
+        fprintf(stderr, "Could not find MultiByteToWideChar entrypoint in coreclr\n");
+        result = 1;
+    }
+    else
+    {
+        setenv("DNX_FRAMEWORK", "dnxcore50", 1);
```

Comment:
Because  the change is just about passing data from native to managed and not about actually using this data (this is why I could pass null in Mono). We are still using the environment variables across the stack since not all bootstrappers pass the data yet (Clr work is in progress and Mono is left). When all bootstrappers pass the struct I will set environment appenvironment properties accordingly and remove all env variables.